sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuRowToColumnarExec.scala

```diff
@@ -289,6 +289,32 @@ private object GpuRowToColumnConverter {
     }
   }
 
+  private class DecimalConverter(precision: Int, scale: Int) extends TypeConverter {
+    override def append(
+        row: SpecializedGetters,
+        column: Int,
+        builder: ai.rapids.cudf.HostColumnVector.ColumnBuilder): Double = {
+      if (row.isNullAt(column)) {
+        builder.appendNull()
+      } else {
+        new NotNullDecimalConverter(precision, scale).append(row, column, builder)
```

Comment:
This is on the data path.  I would like us to avoid object creation if at all possible to speed up the data path.  Please make a static method instead, or use inheritance, which I think is less ideal.